src/pybind/mgr/crash/module.py

```diff
@@ -107,12 +107,22 @@ def handle_command(self, inbuf, command):
         return handler(self, command, inbuf)
 
     def time_from_string(self, timestr):
-        # drop the 'Z' timezone indication, it's always UTC
+        # drop the 'Z' timezone indication: it's always UTC
         timestr = timestr.rstrip('Z')
+        datetime_obj = None
         try:
-            return datetime.datetime.strptime(timestr, DATEFMT)
+            datetime_obj = datetime.datetime.strptime(timestr, DATEFMT)
         except ValueError:
-            return datetime.datetime.strptime(timestr, OLD_DATEFMT)
+            datetime_obj = datetime.datetime.strptime(timestr, OLD_DATEFMT)
+        # add time zone to datetime object before returning it
+        return datetime_obj.replace(tzinfo=datetime.timezone.utc)
```

Comment:
this breaks the `self_test()`.

```
2020-08-18T15:07:02.661 INFO:teuthology.orchestra.run.smithi115.stderr:2020-08-18T15:07:02.654+0000 7fe288575700  1 -- 172.21.15.115:0/1247744208 --> [v2:172.21.15.115:6808/33840,v1:172.21.15.115:6809/338
40] -- mgr_command(tid 0: {"prefix": "mgr self-test module", "module": "crash", "target": ["mon-mgr", ""]}) v1 -- 0x7fe280063560 con 0x7fe258077b50
2020-08-18T15:07:02.662 INFO:teuthology.orchestra.run.smithi115.stderr:2020-08-18T15:07:02.656+0000 7fe26e7fc700  1 -- 172.21.15.115:0/1247744208 <== mgr.4499 v2:172.21.15.115:6808/33840 1 ==== mgr_comman
d_reply(tid 0: -1 Test failed: Remote method threw exception: Traceback (most recent call last):
2020-08-18T15:07:02.662 INFO:teuthology.orchestra.run.smithi115.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 353, in self_test
2020-08-18T15:07:02.662 INFO:teuthology.orchestra.run.smithi115.stderr:    raise RuntimeError('time_from_string() failed')
2020-08-18T15:07:02.662 INFO:teuthology.orchestra.run.smithi115.stderr:RuntimeError: time_from_string() failed
2020-08-18T15:07:02.662 INFO:teuthology.orchestra.run.smithi115.stderr:) v1 ==== 248+0+0 (secure 0 0 0) 0x7fe27c00d410 con 0x7fe258077b50
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:2020-08-18T15:07:02.656+0000 7fe26e7fc700 20 mgrc handle_command_reply tid 0 r -1
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:Error EPERM: Test failed: Remote method threw exception: Traceback (most recent call last):
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 353, in self_test
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:    raise RuntimeError('time_from_string() failed')
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:RuntimeError: time_from_string() failed
2020-08-18T15:07:02.663 INFO:teuthology.orchestra.run.smithi115.stderr:
2020-08-18T15:07:02.664 INFO:tasks.ceph.mgr.x.smithi115.stderr:2020-08-18T15:07:02.656+0000 7f4ae3639700 -1 Remote method threw exception: Traceback (most recent call last):
2020-08-18T15:07:02.664 INFO:tasks.ceph.mgr.x.smithi115.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 353, in self_test
2020-08-18T15:07:02.664 INFO:tasks.ceph.mgr.x.smithi115.stderr:    raise RuntimeError('time_from_string() failed')
2020-08-18T15:07:02.664 INFO:tasks.ceph.mgr.x.smithi115.stderr:RuntimeError: time_from_string() failed
2020-08-18T15:07:02.665 INFO:tasks.ceph.mgr.x.smithi115.stderr:
2020-08-18T15:07:02.665 INFO:tasks.ceph.mgr.x.smithi115.stderr:2020-08-18T15:07:02.656+0000 7f4ae3639700 -1 mgr.server reply reply (1) Operation not permitted Test failed: Remote method threw exception: T
raceback (most recent call last):
2020-08-18T15:07:02.666 INFO:tasks.ceph.mgr.x.smithi115.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 353, in self_test
2020-08-18T15:07:02.666 INFO:tasks.ceph.mgr.x.smithi115.stderr:    raise RuntimeError('time_from_string() failed')
2020-08-18T15:07:02.668 INFO:tasks.ceph.mgr.x.smithi115.stderr:RuntimeError: time_from_string() failed
```

see https://pulpito.ceph.com/kchai-2020-08-18_13:00:33-rados-wip-kefu-testing-2020-08-18-1658-distro-basic-smithi/5356537/

seems it also breaks test in `qa/tasks/mgr/test_insights.py`,

```
2020-08-18T14:50:49.983 INFO:tasks.cephfs_test_runner:test_crash_history (tasks.mgr.test_insights.TestInsights) ... FAIL
2020-08-18T14:50:49.984 INFO:tasks.cephfs_test_runner:
2020-08-18T14:50:49.985 INFO:tasks.cephfs_test_runner:======================================================================
2020-08-18T14:50:49.985 INFO:tasks.cephfs_test_runner:FAIL: test_crash_history (tasks.mgr.test_insights.TestInsights)
2020-08-18T14:50:49.985 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2020-08-18T14:50:49.986 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2020-08-18T14:50:49.986 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-08-18-1658/qa/tasks/mgr/test_insights.py", line 197, in test_crash_history
2020-08-18T14:50:49.986 INFO:tasks.cephfs_test_runner:    self._add_crash(1)
2020-08-18T14:50:49.987 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-08-18-1658/qa/tasks/mgr/test_insights.py", line 45, in _add_crash
2020-08-18T14:50:49.987 INFO:tasks.cephfs_test_runner:    self.assertEqual(0, ret)
2020-08-18T14:50:49.987 INFO:tasks.cephfs_test_runner:AssertionError: 0 != 22
```

see https://pulpito.ceph.com/kchai-2020-08-18_13:00:33-rados-wip-kefu-testing-2020-08-18-1658-distro-basic-smithi/5356503/